Core/SaveState.cpp

```diff
@@ -170,49 +178,69 @@ namespace SaveState
 			return error;
 		}
 
-		void ScheduleCompress(std::vector<u8> *result, const std::vector<u8> *state, const std::vector<u8> *base)
+		void ScheduleCompress(StateBuffer *result, std::vector<u8> *state, const std::vector<u8> *base)
 		{
 			if (compressThread_.joinable())
 				compressThread_.join();
 			compressThread_ = std::thread([=]{
 				SetCurrentThreadName("SaveStateCompress");
 
 				// Should do no I/O, so no JNI thread context needed.
-				Compress(*result, *state, *base);
+				Compress(result, *state, *base);
 			});
 		}
 
-		void Compress(std::vector<u8> &result, const std::vector<u8> &state, const std::vector<u8> &base)
+		void Compress(StateBuffer *result, std::vector<u8> &state, const std::vector<u8> &base)
```

Comment:
The reason is that DoState doesn't take a const pointer, even in read mode, which leaked out. Though, it would make sense to make the outward interface of that const here. I'll clean it up.

Reply:
Actually never mind, this one specifically isn't even affected by that. Not sure why I changed this.